### ISSUE TITLE
Comments on the full documentations

### ISSUE THREAD

**Issue by zhanglw0521** [WilsonGregory/ginjax#173]
The [documentation](https://ginjax.readthedocs.io/) given was easy to read and very thorough. I managed to install the package and run most of the example codes based on the guidelines. I have the following comments:

- I suggest moving the "Math Background" section ahead of "Quick Start" or even combining it with "Home". This will clarify the purpose of the package and ensure that users see the explanations of some terms before they look into the example codes. For instance, the variable `N`, which first appears in "Quick Start", is only explained in "Math Background" but nowhere before. 
- In the "Group and Invariant Filters" section, and in the package, have any groups other than the d-dimensional hyperoctahedral group $B_d$ been implemented? If not, is it planned that the package will be generalized to other groups in the future? If not, I think it should be stated somewhere clearly so that people may have a better idea of what this package is supposed to deal with. 
- When trying to reproduce the examples provided in the documentation, the file 
   '../data/mini_2D_CFD_Rand_M0.1_Eta0.01_Zeta0.01_periodic_128_Train.hdf5'
   seemed not to be provided. I found one without the prefix `mini` under DaRUS (link provided in the documentation), though. Is the mini one a sub-sample of the complete one? 
- After converting a vanilla model to its equivariant version, can the authors comment on how the number of parameters changes, please? Intuitively, I expect a decrease in the numbers because of the additional equivariance constraints, and hence the model is smaller but allows better accuracy. If this is the case, it might be worth commenting somewhere, either in the paper or in the example section. On a similar note, however, I also see: 
   ```
   Equivariant Model (126905), 0.001
   ...
   Vanilla Model (87940), 0.001
   ...
   ```
    on the documentation (example 1). Do the two numbers, 126905 and 87940, mean the number of parameters? If so, why is there an increase in the number of parameters? 

- Would adding colorbars that indicate the magnitude of errors to the figures under blocks [11] and [12] of Example 1 make things clearer? 
- Additionally, it was mentioned that "Since our training data set is so small, the predictions don't look great". My thought was that a smaller training data set would, in general, give a smaller training error as well, and if the predictions don't look great, it is probably that the model isn't rich enough. I understand that the error is not important at all in a tutorial example, but the sentence seemed a bit nonintuitive to me.

Reference: https://github.com/openjournals/joss-reviews/issues/8129

**Comment by WilsonGregory** [comment]
Thank you so much for the detailed feedback. Here are our responses to each point:

> I suggest moving the "Math Background" section ahead of "Quick Start" or even combining it with "Home". This will clarify the purpose of the package and ensure that users see the explanations of some terms before they look into the example codes. For instance, the variable N, which first appears in "Quick Start", is only explained in "Math Background" but nowhere before.

We have expanded the description of geometric images in "Home" to highlight how they behave under rotations, and we have added the "Equivariance" section to give a brief introduction to the groups and equivariance that are covered more fully in those separate pages. This should hopefully communicate the main motivation and idea before the users get to Quickstart. I like Quickstart to still be second because I think it is helpful to be able to play around with the objects before jumping back into the theory. We added a description of N to the quickstart page.

> In the "Group and Invariant Filters" section, and in the package, have any groups other than the d-dimensional hyperoctahedral group B d been implemented? If not, is it planned that the package will be generalized to other groups in the future? If not, I think it should be stated somewhere clearly so that people may have a better idea of what this package is supposed to deal with.

Yes, to avoid voxelization and approximate equivariance, we will stick with the hyperoctahedral group or one of its subgroups. We have updated the page to explain this.

> When trying to reproduce the examples provided in the documentation, the file
'../data/mini_2D_CFD_Rand_M0.1_Eta0.01_Zeta0.01_periodic_128_Train.hdf5'
seemed not to be provided. I found one without the prefix mini under DaRUS (link provided in the documentation), though. Is the mini one a sub-sample of the complete one?

Hmm, I am unable to reproduce this issue. The mini data set is in the repo here: [https://github.com/WilsonGregory/ginjax/tree/main/data](https://github.com/WilsonGregory/ginjax/tree/main/data). It is a mini version of that larger dataset with only 4 trajectories (22 Mb) instead of 10,000 (53 Gb). I added a note to update the relative path if necessary.

> After converting a vanilla model to its equivariant version, can the authors comment on how the number of parameters changes, please? Intuitively, I expect a decrease in the numbers because of the additional equivariance constraints, and hence the model is smaller but allows better accuracy. If this is the case, it might be worth commenting somewhere, either in the paper or in the example section. On a similar note, however, I also see:

> Equivariant Model (126905), 0.001
>Vanilla Model (87940), 0.001

> on the documentation (example 1). Do the two numbers, 126905 and 87940, mean the number of parameters? If so, why is there an increase in the number of parameters?

Thank you for bringing this up, this is a good point. Those numbers are indeed the number of parameters. The answer to your question turns out to be rather involved, and the full answer is in the larger paper, tables 1 and 2: [https://royalsocietypublishing.org/doi/full/10.1098/rsta.2024.0247#d1e11735](https://royalsocietypublishing.org/doi/full/10.1098/rsta.2024.0247#d1e11735).

It comes down to how model "width" is defined for the GeometricImageNet hidden layers. For a standard CNN, width is the number of channels, so 32 in the conversion example. For a GI-Net, width is the number of channels for each image type, which for the conversion example has both scalar and vector images. So there are 32 scalar channels and 32 vector channels (which each have 2 components). This could be viewed as 32 + 32*2 = 96 channels, although the vector components are not independent so its not exactly equivalent to 96 channels of a vanilla CNN. So although each filter image has fewer parameters because of the equivariant restrictions, the higher number of channels leads to a higher overall number of parameters.

In the main paper, we balance this out by having 2 model comparisons:
1) reduce the number of channels of the equivariant CNN so that the overall number of parameters is roughly equal to the vanilla CNN, and
2) reduce the number of channels so that for example a width 64 vanilla CNN is compared to a width 20 equivariant CNN because 20 + 20*2 = 60 which is close to 64 channels.

In almost cases, these smaller equivariant CNN models still outperform the vanilla CNN models. However, for the purposes of the simple conversion example we decided to just go with the straight forward option of keeping the width the same. But this is an important feature to explain, so we have added a discussion in the conversion example.

> Would adding colorbars that indicate the magnitude of errors to the figures under blocks [11] and [12] of Example 1 make things clearer?

Good idea, we have added colorbars.

> Additionally, it was mentioned that "Since our training data set is so small, the predictions don't look great". My thought was that a smaller training data set would, in general, give a smaller training error as well, and if the predictions don't look great, it is probably that the model isn't rich enough. I understand that the error is not important at all in a tutorial example, but the sentence seemed a bit nonintuitive to me.

That is a good point, it should be easier to overfit to 0 training error with a small data set. In this case, we use the same number epochs that we do for the larger data set, but since each epoch is much smaller it results in fewer batch steps being taken. We changed the line to, "For this small example we just train long enough to get reasonable looking results, but hopefully this gives you an idea of the problem."

See https://github.com/WilsonGregory/ginjax/pull/178 for fixes.

**Comment by zhanglw0521** [comment]
> Hmm, I am unable to reproduce this issue. The mini data set is in the repo here: https://github.com/WilsonGregory/ginjax/tree/main/data. It is a mini version of that larger dataset with only 4 trajectories (22 Mb) instead of 10,000 (53 Gb). I added a note to update the relative path if necessary.

Ah, my bad. It was my routine setting that caused the problem. In my case, removing `../`, or simply going into the `examples` folder helps to resolve the issue I had. I guess there's nothing to be changed here. 

> Good idea, we have added colorbars.

Would it make more sense if the colorbars were scaled more consistently? It would be clearer to me if the four figures for predictions and targets, and the two figures for differences, could have the same scaling, respectively, as that would make it easier to compare the two models directly. 


As for the number of parameters, I now agree that it is a bit tricky to make such a full explanation in the documentation, and I find the current one, as an example, persuasive.

**Comment by WilsonGregory** [comment]
Your suggestion for the colorbars makes sense, here are the updated version:

<img width="926" height="615" alt="Image" src="https://github.com/user-attachments/assets/866ab782-e491-437a-8fb0-5aafe26716d9" />
and 

<img width="909" height="640" alt="Image" src="https://github.com/user-attachments/assets/c8997174-a629-4829-9cb1-3742484ea571" />
